codex-rs/login/src/server.rs

```diff
@@ -89,11 +118,35 @@ pub fn run_login_server(
     }
     let shutdown_flag = shutdown_flag.unwrap_or_else(|| Arc::new(AtomicBool::new(false)));
     let shutdown_flag_clone = shutdown_flag.clone();
+    let timeout_flag = Arc::new(AtomicBool::new(false));
+
+    // Channel used to signal completion to timeout watcher.
+    let (done_tx, done_rx) = mpsc::channel::<()>();
```

Comment:
I was going to suggest this might be a candidate for `std::sync::mpsc::oneshot`, but it looks like there are two places where `done_tx.send()` could be called, so I guess this is not a "single producer" case?